velox/exec/HashBuild.cpp

```diff
@@ -51,24 +95,87 @@ void HashJoinBridge::setAntiJoinHasNullKeys() {
 std::optional<HashJoinBridge::HashBuildResult> HashJoinBridge::tableOrFuture(
     ContinueFuture* future) {
   std::lock_guard<std::mutex> l(mutex_);
+  VELOX_DCHECK(started_);
+  VELOX_CHECK(!cancelled_, "Getting hash table after join is aborted");
   VELOX_CHECK(
-      !cancelled_, "Getting hash table after the build side is aborted");
-  if (table_ || antiJoinHasNullKeys_) {
-    return HashBuildResult{table_, antiJoinHasNullKeys_};
+      !buildResult_.has_value() ||
+      !restoringSpillPartitionId_.has_value() && restoringSpillShards_.empty());
+
+  if (buildResult_.has_value()) {
+    return buildResult_.value();
   }
   promises_.emplace_back("HashJoinBridge::tableOrFuture");
   *future = promises_.back().getSemiFuture();
   return std::nullopt;
 }
 
+bool HashJoinBridge::setSpillInput() {
+  std::vector<ContinuePromise> promises;
+  bool hasSpillInput = false;
+  {
+    std::lock_guard<std::mutex> l(mutex_);
+    VELOX_DCHECK(started_);
+    VELOX_CHECK(buildResult_.has_value());
+    VELOX_CHECK(
+        !restoringSpillPartitionId_.has_value() &&
+        restoringSpillShards_.empty());
+    VELOX_CHECK_GT(numBuilders_, 0);
+
+    buildResult_.reset();
```

Comment:
Here, we are clearing the hash table as it has been fully processed and not needed anymore. We'll wait for the HashBuild operator to build a new table from the next spill partition now.